addons/pharma/functions/fnc_init.sqf

```diff
@@ -117,50 +117,212 @@ if (GVAR(kidneyAction)) then {
+            private _logString = LSTRING(coagulation_Bandaged);
+            private _woundClotTime = 0;
+            private _bandageToUse = "";
+            private _coagulation_time_minor = missionNamespace getVariable [QGVAR(coagulation_time_minor), 15];
+            private _coagulation_time_medium = missionNamespace getVariable [QGVAR(coagulation_time_medium), 30];
+            private _coagulation_time_large = missionNamespace getVariable [QGVAR(coagulation_time_large), 45];
+            private _factorCoutToRemove = 1;
+
+            switch (_suffix) do {
+                case "Minor": { 
+                    _woundClotTime = round ((random (_coagulation_time_minor / 2)) + _coagulation_time_minor / 2);
+                    _bandageToUse = "BloodClotMinor";
+                    _factorCoutToRemove = 1;
+
+                    if !(missionNamespace getVariable [QGVAR(coagulation_allow_MinorWounds), true]) then { continue; };
+                };
+                case "Medium": { 
+                    _woundClotTime = round ((random (_coagulation_time_medium / 2)) + _coagulation_time_medium / 2);
+                    _bandageToUse = "BloodClotMedium";
+                    _factorCoutToRemove = 1.5;
+
+                    if !(missionNamespace getVariable [QGVAR(coagulation_allow_MediumWounds), true]) then { continue; };
+                };
+                default { 
+                    _woundClotTime = round ((random (_coagulation_time_large / 2)) + _coagulation_time_large / 2);
+                    _bandageToUse = "BloodClotLarge";
+                    _factorCoutToRemove = 2;
+
+                    if !(missionNamespace getVariable [QGVAR(coagulation_allow_LargeWounds), true]) then { continue; };
+                };
+            };
+
+            if (_amountOf * _bleeding > 0) exitWith {
+
+                if (_countTXA > 0 || _countEACA > 0) then {
+                    _logString = LSTRING(coagulation_Bandaged_TXA);
+
+                    if (_countEACA > 0 && _countTXA > 0) exitWith { // If TXA & EACA are in system at same time use EACA bandage
+                        _woundClotTime = round (_woundClotTime / 3 * 2);
+                        _bandageToUse = _bandageToUse + "EACA";
+                    };
+
+                    _bandageToUse = _bandageToUse + "TXA";
+                };
+
+                [{
+                    params["_unit", "_bodyPart", "_selectionName", "_bandageToUse", "_logString", "_factorCoutToRemove"];
+
+                    private _coagulationFactor = _unit getVariable [QGVAR(coagulationFactor), 15];
+                    private _openWounds = GET_OPEN_WOUNDS(_unit);
+                    private _openWoundsOnPart = _openWounds getOrDefault [_bodyPart, []];
+                    private _woundIndex = _openWoundsOnPart findIf {(_x select 1) > 0 && (_x select 2) > 0};
+
+                    if (_coagulationFactor <= 0) exitWith {};
+                    if (_woundIndex == -1) exitWith {};
```

Comment:
there is another check before wound already, this is inside WaitAndExecute so if there was another wound closed and we now have no factors left but due to CBA WaE this call is still initialized it will make call on body part even tho there are no factors left.